web/packages/teleport/src/Navigation/SideNavigation/Search.tsx

```diff
@@ -0,0 +1,204 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useRef, useEffect } from 'react';
+import { NavLink } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Box, Flex, Text } from 'design';
+import { height, space, color } from 'design/system';
+
+import { NavigationSection, NavigationSubsection } from './Navigation';
+import {
+  Section,
+  RightPanel,
+  verticalPadding,
+  getSubsectionStyles,
+} from './Section';
+import { CategoryIcon } from './CategoryIcon';
+import { CustomNavigationCategory } from './categories';
+
+export function SearchSection({
+  navigationSections,
+  expandedSection,
+  previousExpandedSection,
+  handleSetExpandedSection,
+  currentView,
+}: {
+  navigationSections: NavigationSection[];
+  expandedSection: NavigationSection;
+  previousExpandedSection: NavigationSection;
+  currentView: NavigationSubsection;
+  handleSetExpandedSection: (section: NavigationSection) => void;
+}) {
+  const section: NavigationSection = {
+    category: CustomNavigationCategory.Search,
+    subsections: [],
+  };
+
+  const isExpanded =
+    expandedSection?.category === CustomNavigationCategory.Search;
+  return (
+    <Section
+      key="search"
+      section={section}
+      $active={false}
+      onClick={() => null}
+      setExpandedSection={() => handleSetExpandedSection(section)}
+      aria-controls={`panel-${expandedSection?.category}`}
+      isExpanded={isExpanded}
+    >
+      <RightPanel
+        isVisible={isExpanded}
+        skipAnimation={!!previousExpandedSection}
+        id={`panel-${section.category}`}
+        onFocus={() => handleSetExpandedSection(section)}
+      >
+        <SearchContent
+          navigationSections={navigationSections}
+          isVisible={isExpanded}
+          currentView={currentView}
+        />
+      </RightPanel>
+    </Section>
+  );
+}
+
+function SearchContent({
+  navigationSections,
+  isVisible,
+  currentView,
+}: {
+  navigationSections: NavigationSection[];
+  isVisible: boolean;
+  currentView: NavigationSubsection;
+}) {
+  const [searchInput, setSearchInput] = useState('');
+  const [results, setResults] = useState<NavigationSubsection[]>([]);
+  const inputRef = useRef<HTMLInputElement>();
+
+  useEffect(() => {
+    if (isVisible) {
+      setSearchInput('');
+      setResults([]);
+    }
+  }, [isVisible]);
+
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
I think this should handle a bit less. this `handleSearch` should really just be setting the state of the search value.

Filtered results can just be derived on each render instead of being held in state. something like this
copy and then `pbpaste | git apply`

```
diff --git a/web/packages/teleport/src/Navigation/SideNavigation/Search.tsx b/web/packages/teleport/src/Navigation/SideNavigation/Search.tsx
index ffe6c7355d..323603ac5b 100644
--- a/web/packages/teleport/src/Navigation/SideNavigation/Search.tsx
+++ b/web/packages/teleport/src/Navigation/SideNavigation/Search.tsx
@@ -89,34 +89,26 @@ function SearchContent({
   currentView: NavigationSubsection;
 }) {
   const [searchInput, setSearchInput] = useState('');
-  const [results, setResults] = useState<NavigationSubsection[]>([]);
   const inputRef = useRef<HTMLInputElement>();
 
   useEffect(() => {
     if (isVisible) {
       setSearchInput('');
-      setResults([]);
     }
   }, [isVisible]);
 
   const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const lowerCaseInput = e.target.value.toLowerCase();
-    setSearchInput(lowerCaseInput);
-
-    if (lowerCaseInput.length > 0) {
-      const searchResults = navigationSections.flatMap(section =>
-        section.subsections.filter(subsection =>
-          subsection.searchableTags?.some(tag =>
-            tag.toLowerCase().includes(lowerCaseInput)
-          )
-        )
-      );
-      setResults(searchResults);
-    } else {
-      setResults([]);
-    }
+    setSearchInput(e.target.value);
   };
 
+  const results = navigationSections.flatMap(section =>
+    section.subsections.filter(subsection =>
+      subsection.searchableTags?.some(tag =>
+        tag.toLowerCase().includes(searchInput.toLocaleLowerCase())
+      )
+    )
+  );
+
   return (
     <Flex flexDirection="column">
       <Flex py={verticalPadding} px={3}>

```

also, storing "raw" input is better than mutating it before storage (in this case, as lowercase) because it may throw users off. i know the case doesnt matter in this search, but i expect to see exactly what i type in the field